src/posts/2021-09-27-Using-focus-style-for-accessibility

```diff
@@ -0,0 +1,119 @@
+    year: 2016
+  - title: never-remove-css-outlines
+    url: https://www.a11yproject.com/posts/2013-01-25-never-remove-css-outlines/
+    source: Guilherme Simões
+    year: 2013
+tags:
+  - How-to
+
+---
+
+While accessing any digital interface, a focus indicator is important it highlights the active intractable component in a webpage. 
+
+## What is a focus style?
+A focus is a pseudo-class used to indicate or highlights an intractable component in a webpage.
+
+## Why is it important?
+A focus style is necessary because some people use a mouse or a touch device to access a website some rely on using a keyboard to navigate on a website. Having a focus indicator will improve the usability of your website and makes it accessible, It also tells us the interactive element we are focused on.
+
+#HowtoApplyFocusStyle
+
+Below is an example of how we can apply focus style to our elements.
+
+## Applying focus style to a link
+
+```CSS
+a:focus {
+  outline: 5px solid red;
+  }
+```
+The above code example is a link that has a custom style for its focus state.
+
+**Code expalnation**
+In the above example, we added focus to our link and gave it an outline of solid red with a value of 5px. Please note the focused style is applied only to the link tag itself.
+
+The code below is another example for applying focus.
+
+## Applying focus style to a form input
+
+```HTML
+<form>
+<label for="email"> Email :</label>
+<input class="email-input" value="Click me to see my focus state">
+</form>
+```
+```CSS
+.email-input:focus {
+ background-color: gray;
+ color: black;
+}
+```
+You can create your custom outline style aside from the blue outline default style in focus. Always ensure the color contrast for your focus indicator is sufficient especially important for people with low vision, this will also benefit those using their screen outside in a very bright space.
+
+## Focus-visible
+Is a great idea for controlling how users interact with your elements it is used to provide a focus-indicator selectively. 
+
+Let’s look at a quick code example
+
+```HTML
+<form class="form-container">
+<label for="email">Email:</label>
+<input class="email-input" value="Click me to see my focus state">
+<label for="username">Username:</label>
+<input class="user-input" value="Click me to see my focus state">
+<button type="submit">Submit</button>
+</form>
+```
+```CSS
+:focus {
+ outline: 2px solid orange
+}
+```
+In the above code when we focus on each of the input and buttons it gives us a red outline but focus style can also affect the user experience of mouse users.
+
+It is wrong to remove the focus indicator as it makes your website inaccessible for people navigating with the keyboard the best way is to apply a different focus style for mouse/pointer users.
+
+code example
+```CSS
+:focus:not(:focus-visible) {
+ outline:none
+}
+```
+This will hide the focus indicator if the element receives focus via the mouse, but it will show up on the keyboard when tabbing.
```

Comment:
```suggestion

This will hide the focus indicator if the element receives focus via the mouse, but it will show up on the keyboard when tabbing.
```

Reply:
Thanks @ericwbailey  i have fixed the required changes please kindly review